src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/runInTerminalTool.ts

```diff
@@ -297,14 +297,35 @@ export class RunInTerminalTool extends Disposable implements IToolImpl {
 				if (!outputAndIdle.terminalExecutionIdleBeforeTimeout) {
 					outputAndIdle = await racePollingOrPrompt(
 						() => pollForOutputAndIdle(execution, true, token, this._languageModelsService),
-						() => promptForMorePolling(command, invocation.context!, this._chatService),
+						() => promptForMorePolling(localize('poll.terminal.waiting', "Continue waiting for `{0}` to finish?", command), localize('poll.terminal.polling', "Copilot will continue to poll for output to determine when the terminal becomes idle for up to 2 minutes."), invocation.context!, this._chatService),
 						outputAndIdle,
 						token,
 						this._languageModelsService,
 						execution
 					);
 				}
-
+				const userInputKind = await getExpectedUserInputKind(outputAndIdle.output);
```

Comment:
This could happen multiple times and also maybe only after some period of time has passed, I think it needs to live inside the polling loop which should be doing general evaluation of the state instead of just detecting idle.